components/admin_console/team_channel_settings/team/details/team_details.jsx

```diff
@@ -179,7 +214,40 @@ export default class TeamDetails extends React.Component {
                 serverError = ex;
             }
         }
-        this.setState({groups, usersToRemove, saveNeeded: true, serverError});
+        this.setState({groups, usersToRemoveCount, saveNeeded: true, serverError});
+    }
+
+    addUsersToAdd = (users) => {
+        let {usersToRemoveCount} = this.state;
+        const {usersToAdd, usersToRemove} = this.state;
+        users.forEach((user) => {
+            if (usersToRemove[user.id]?.id === user.id) {
+                delete usersToRemove[user.id];
+                usersToRemoveCount -= 1;
+            } else {
+                usersToAdd[user.id] = user;
+            }
+        });
+        this.setState({usersToAdd: {...usersToAdd}, usersToRemove: {...usersToRemove}, usersToRemoveCount, saveNeeded: true});
+    }
+
+    addUserToRemove = (user) => {
+        let {usersToRemoveCount} = this.state;
+        const {usersToAdd, usersToRemove, rolesToUpdate} = this.state;
+        if (usersToAdd[user.id]?.id === user.id) {
+            delete usersToAdd[user.id];
+        } else if (usersToRemove[user.id]?.id !== user.id) {
+            usersToRemoveCount += 1;
```

Comment:
Actually im gonna `usersToRemoveCount` since i'm just using the same modal that is used for confirming when you sync groups right now maybe I can refactor this later if we need two different modals but right now it makes sense to just use the same variable.